cmd/harmony/main.go

```diff
@@ -317,12 +380,14 @@ func setupConsensusAndNode(nodeConfig *nodeconfig.ConfigType) *node.Node {
 	currentConsensus.Decider.SetShardIDProvider(func() (uint32, error) {
```

Comment:
we should refactor this dependency injection into a proper interface, not cbs

Reply:
can you elaborate?